server/text_generation_server/utils/layers.py

```diff
@@ -713,11 +727,16 @@ def forward(
 
                 # Inplace operation, updating query and key.
                 pos_encoding_ops.rotary_embedding(query, key, head_size, cos, sin, True)
+            elif IS_XPU_SYSTEM:
```

Comment:
This could be moved to get_cos_sin to avoid launching too many unnecessary kernels

Reply:
the func is to use query and key to generate the cos and sin. get_cos_sin is just to get the generated cos and sin. would you like to combine this two?

Reply:
Olviier meant it should be done directly here: https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/utils/layers.py#L838

That avoids creating this repeat over and over.

Reply:
ok, got it, thanks